src/restore.js

```diff
@@ -25,156 +26,64 @@ const visit = require("unist-util-visit");
-    //
-    // A redacted block will actually be two "blocks" representing the open and
-    // close of a redacted block. Together, they will look like:
-    //
-    //     [some text][0]
-    //
-    //     some other markdown content contained within the redaction
-    //
-    //     [/][0]
-    //
-    // Where "some text" is something that can be translated  and "0" is the
-    // index of the redacted value.
-    const tokenizeBlockRedaction = function (eat, value, silent) {
-      const BLOCK_REDACTION_RE = /^\[([^\]]*)\]\[(\d+)\]\n\n/;
-      const startMatch = BLOCK_REDACTION_RE.exec(value)
-
-      // if we don't find an open block, return immediately
-      if (!startMatch) {
-        return;
-      }
-
-      // the entire string representing the "open" block of the redaction
-      const blockOpen = startMatch[0];
-
-      // the index within `value` at which the inner content of the redacted block begins
-      // (ie the index at which the "open" block ends)
-      const startIndex = startMatch[0].length;
-
-      // the translated data inside the first set of brackets
-      const content = startMatch[1];
-
-      // the sequential index inside the second set of brackets
-      const index = parseInt(startMatch[2], 10);
-
-      // if we don't have a redaction matching this index, return immediately
-      // TODO once we decide on how we want to handle errors, this is where the
-      // error handler should probably go
-      const redactedData = redactions[index];
-      if (!(redactedData && redactedData.type === "blockRedaction")) {
-        return;
-      }
-
-      // the entire string representing the "close" block of the redaction
-      const blockClose = `\n\n[/][${index}]`;
-
-      // the index within `value` at which the inner content of the redacted block ends
-      // (ie the index at which the "close" block begins)
-      const endIndex = value.indexOf(blockClose, startIndex);
-
-      // if we didn't find a close block, return immediately
-      if (endIndex === -1) {
-        return;
-      }
-
-      const restorationMethod = Parser.prototype.restorationMethods[redactedData.redactionType];
-      if (!restorationMethod) {
-        return
-      }
-
-      // if we get to here, then we have found a valid block! Return true if in
-      // silent mode to indicate a passing test
-      if (silent) {
-        return true;
-      }
-
-      // if in normal (ie non-silent) mode, consume the token and produce a
-      // render
-      const subvalue = value.slice(startIndex, endIndex);
-      const children = this.tokenizeBlock(subvalue, eat.now());
-      const add = eat(blockOpen + subvalue + blockClose);
-      return restorationMethod(add, redactedData, content, children);
+    const restorationMethod = restorationMethods[redactedData.redactionType];
+    if (restorationMethod) {
+      const restored = restorationMethod(redactedData, node.content);
+      Object.assign(node, restored);
```

Comment:
why this instead of `parent.children.splice(index, 1, restored)`?

Reply:
Changed. Splicing in the node is definitely better.